src/HttpAsyncClientDiscovery.php

```diff
@@ -24,7 +24,7 @@ public static function find()
         try {
             $asyncClient = static::findOneByType(HttpAsyncClient::class);
 
-            return new $asyncClient();
+            return static::instantiateClass($asyncClient);
```

Comment:
Then maybe we should move the instantiation out of the try-catch block. Makes the code easier to read IMO.